crates/daphne-server/src/router/mod.rs

```diff
@@ -390,9 +480,28 @@ mod test {
             fn signing_key(&self) -> Option<&p256::ecdsa::SigningKey> {
                 None
             }
+
+            async fn check_bearer_token(
+                &self,
+                token: &BearerToken,
+                _sender: DapSender,
+                _task_id: TaskId,
+                _is_taskprov: bool,
+            ) -> Result<(), Either<String, DapError>> {
+                (token.as_str() == BEARER_TOKEN)
+                    .then_some(())
+                    .ok_or_else(|| Left("invalid token".into()))
+            }
         }
 
         async fn handler(
+            State(ch): State<Arc<Channel>>,
+            UnauthenticatedDapRequestExtractor(req): UnauthenticatedDapRequestExtractor,
+        ) -> impl IntoResponse {
+            ch.send(req).await.unwrap();
```

Comment:
Hmm, the return type of this function is `()`. Is that intended? Is this unwrap() safe?

Reply:
Yes, this is fine. We are actually just "smugling" the request out through the channel so we can observe in the test how it was parsed

Reply:
Add a comment to explain.